File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int l=0;
        int r=height.size()-1;
        int maxWater=0;
        while(l<r){
            int w=r-l;
            int h=min(height[l],height[r]);
            maxWater=max(w*h,maxWater);
            height[l]<height[r]?l++:r--;
        }
        return maxWater;
    }
};

File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int finalArea=0;
        int currentArea=0;
        int l=0, r=height.length-1;
        while(l<r){
            currentArea=Math.min(height[l],height[r])*(r-l);
            finalArea=Math.max(currentArea,finalArea);
            if(height[l]>height[r]) r--;
            else l++;
        }
        return finalArea;
    }
}